onnxruntime/core/providers/webgpu/buffer_manager.cc

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/webgpu/buffer_manager.h"
+#include "core/providers/webgpu/webgpu_context.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+size_t NormalizeBufferSize(size_t size) {
+  return (size + 15) / 16 * 16;
+}
+
+class DisabledCacheManager : public IBufferCacheManager {
+  size_t CalculateBufferSize(size_t request_size) override {
+    return NormalizeBufferSize(request_size);
+  }
+
+  WGPUBuffer TryAcquireCachedBuffer(size_t /*buffer_size*/) override {
+    // always return empty buffer
+    return nullptr;
+  }
+  void RegisterBuffer(WGPUBuffer /*buffer*/, size_t /*request_size*/) override {
```

Comment:
>RegisterBuffer [](http://example.com/codeflow?start=7&length=14)

Why RegisterBuffer is a no-op, but ReleaseBuffer is not?Why the no-op implementations are needed here?

Reply:
`IBufferCacheManager` is designed to make it possible to inform the actual implementation all events about the buffer.

Later, I found the current 4 implementation do not use it. It is still possible that a new cache manager may want to be notified is a buffer is actually created, depending on how the cache policy is.